apps/pipeline/lib/pipeline/reader/dataset_topic_reader.ex

```diff
@@ -0,0 +1,56 @@
+defmodule Pipeline.Reader.DatasetTopicReader do
```

Comment:
It's on purpose. Both functions require a `dataset` argument, where neither `TopicWriter` or `TableWriter` does.

Reply:
By that logic shouldn't the writer be named `ProducerTopicWriter`?
The current `TopicWriter` also reads like it's doing ad hoc connections to produce data to topics, which isn't going to optimal; since our most recent issues with failure have been the loss of brod producers and the latest Elsa centralizes supervision of both producers and consumers under a single supervisor, i'm not sure you're going to get the best performance if you don't assume a similar `DatasetTopicWriter` and ensure a persistent, dedicated producer supervised for the life of that dataset/ingestion event.